Update: SDK demo site account creation / login page

jennsun opened the pull request:
https://app.clickup.com/t/yqfjyv

Updated account creation/login page of the SDK Demo Site. Changed layout to include proper titles/subtitles, field order, and button style (theBlue color).

jennsun committed:
Update: SDK Demo Site Account Creation/Login page

simanthastew commented on example_site/src/components/AccountInfoForm.js:
```diff
@@ -26,7 +25,34 @@ export default function AccountInfoForm({
       onSubmit={handleSubmit(onSubmit, onFormSubmitError)}
     >
-      <h2>{formTitle}</h2>
+      <h2>Create new account</h2>
```
This component is used on the update account page, so we won't to display "Create new account" - I would revert back to using the `formTitle` prop. And since we're adding the "All fields are optional" text, maybe an optional `subtitle` prop 👀 

For optional props, you'll need to make sure that the component looks and behaves correctly with and without it!

simanthastew commented on example_site/src/components/AccountAuthentication.css:
```diff
@@ -52,3 +52,29 @@
+  text-align: center;
+  width: 40%;
+  padding: 10px;
+}
```
It looks like this is identical to the class in `ConfigurationControls.css` and is used to wrap the `Header` component. You should be able to use these styles in the `Header.css` file and apply them globally to the `Header` instead of defining this class in each individual stylesheet.

simanthastew commented on example_site/src/components/AccountAuthentication.css:
```diff
@@ -52,3 +52,29 @@
+  grid-template-columns: repeat(2, 1fr);
+}
+
+.center-content{
+  margin-left: 200px;
+  margin-right: 200px;
+}
+
+.center-left{
+  margin-left: 400px;
+}
+
+.center-right{
+  margin-right: 350px;
+}
```
These are some pretty big margins, and when I load up the site we've got some extra horizontal scrolling going on. When you give margins with set amount of pixels like that, the browser will expand the space to take up as many pixels as you've defined. I think you need to add some modest padding/margin to the `<main>` tag in `App.js` that will apply to the entire site, and rely on that to keep the margins consistent. Or, we could give the `main` tag a max width with some padding.  
https://share.getcloudapp.com/OAuQ4Pnj

simanthastew replied:
Additionally, we should be able to control how those forms are rendered inside of the `.two-columns` class using different grid layout attributes.

simanthastew commented on example_site/src/components/App.css:
```diff
@@ -113,3 +108,7 @@ div.loading {
   margin-left: 20px;
 }
+
+.move-up {
+  margin-top: -25px;
+}
```
We should not need this (although it's a clever solution!).

The div this is applied to also has another class, `panel`. `panel` has 24px of padding applied to it, so that explains the -25px margin here. The `panel` class is leftover from the old site layout, so I would get rid of it here, or look around at other places its being applied to see if it's really relevant anymore.

simanthastew reviewed (changes requested):
Nice work! I think we can simplify a lot of the styling on this one, and get a few good layout principles enacted for the entire site that will make the rest of the work a little easier.

jennsun committed:
PR Review
- Reverted to using formTitle prop
- Condensed a lot of the CSS to be more site-wide instead of local, reduced margin scrolling issues

simanthastew commented on example_site/src/components/AccountAuthentication.js:
```diff
@@ -144,21 +147,29 @@ function CreateAccountForm() {
-      <CreateAccountForm />
-      <LoginAccountForm />
-    </>
+    <div>
```
You can use a fragment here since this `div` is unstyled

simanthastew commented on example_site/src/components/App.css:
```diff
@@ -35,38 +36,30 @@ main {
+  padding-right: 40px;
+}
+
 .submit-button {
```
Since we want all buttons to be uppercase, instead of setting text in JS as uppercase, i.e. `SEE HOW IT WORKS`, you can add `text-transform: uppercase;` here. This way the rule is enforced site-wide without you or future developers needing to remember to use uppercase.

simanthastew commented on example_site/src/components/AccountAuthentication.js:
```diff
@@ -74,38 +76,39 @@ function LoginAccountForm() {
 
   return (
-    <div className="panel">
+    <div>
```
You don't need this `div` at all now - everything is contained within the `form` element!

simanthastew commented on example_site/src/components/AccountAuthentication.js:
```diff
@@ -144,21 +147,29 @@ function CreateAccountForm() {
+        you can create a Waymark account with any information you
+        use to identify users on your end."
+      /> 
+      <p></p>
```
Why the empty `<p>` tag?

jennsun replied:
Initially used for spacing - changed so top padding was included in two-columns class

simanthastew commented on example_site/src/components/AccountAuthentication.js:
```diff
@@ -144,21 +147,29 @@ function CreateAccountForm() {
+        use to identify users on your end."
+      /> 
+      <p></p>
+      <div className='two-columns left'>
```
`two-columns` and `left` are both only being used here so you can combine those classes in the stylesheet!

simanthastew commented on example_site/src/components/AccountAuthentication.js:
```diff
@@ -74,38 +76,39 @@ function LoginAccountForm() {
 
-        <button className="submit-button" data-test="loginAccount-button">
-          Login
+        <button className="submit-button form-input" data-test="loginAccount-button">
```
Are you just applying the`form-input` class to get the correct width?

Since this is a button (not a form input) I would define a separate class in `App.css` called `form-button` that you can apply here instead.

```
form-input,
form-button {
    width: 20em;
}
```

simanthastew commented on example_site/src/components/AccountInfoForm.js:
```diff
@@ -118,19 +136,7 @@ export default function AccountInfoForm({
-      />
-
-      <button className="submit-button" data-test="createAccount-button">
+      <button className="submit-button form-input" data-test="createAccount-button">
```
You can use the new `form-button` class here as well.

simanthastew commented on example_site/src/components/ConfigurationControls.js:
```diff
@@ -183,7 +183,7 @@ export default function ConfigurationControls({ isOpen }) {
   return (
     <form onSubmit={handleSubmit(onSubmit)} className={titlePanel}>
-      <div className='center-title'>
+      <div>
```
You shouldn't need this `div` anymore

simanthastew commented on example_site/src/components/Header.css:
```diff
@@ -1,5 +1,11 @@
+.center {
```
Typically a classname like this is something that's shared among a lot of different elements because it's just conveying that they should be centered.

Since this class specifically applies to the header, let's give it a name that ties it to the component, like `HeaderContainer`

simanthastew replied:
This doesn't matter since we're getting rid of the `center` class in `App.css`, but remember that everything inherits from `App`. You see how you can use the `form-input` class in any component even though it's defined in `App.css`? The same could have been said for the `center` class so it's a good idea to avoid class name collisions like that.

simanthastew commented on example_site/src/components/App.css:
```diff
@@ -35,38 +36,30 @@ main {
   margin: 12px 0 4px;
 }
 
+.center {
```
You can just add these styles to the `main` section above instead of defining a new class

simanthastew reviewed (changes requested):
This is looking really good - just a few really small updates to make. Feel free to merge when you're done!

jennsun committed:
PR Review 2
-Removed divs/classes where unnecessary
-Applied uppercase transform to all button styles so styling could be enforced without manually typing in uppercase name every time
-Combined classes where possible
-Renamed class names to avoid class name collisions
